server/channels/api4/custom_profile_attributes_test.go

```diff
@@ -0,0 +1,247 @@
+	})
+
+	th.TestForSystemAdminAndLocal(t, func(t *testing.T, client *model.Client4) {
+		field := &model.PropertyField{
+			Name: model.NewId(),
+			Type: model.PropertyFieldTypeText,
+		}
+		createdField, appErr := th.App.CreateCPAField(field)
+		require.Nil(t, appErr)
+		require.NotNil(t, createdField)
+
+		patch := &model.PropertyFieldPatch{Name: model.NewPointer(model.NewId())}
+		patchedField, resp, err := client.PatchCPAField(context.Background(), createdField.ID, patch)
+		CheckOKStatus(t, resp)
+		require.NoError(t, err)
+		require.Equal(t, *patch.Name, patchedField.Name)
+	}, "a user with admin permissions should be able to patch the field")
+}
+
+func TestDeleteCPAField(t *testing.T) {
+	os.Setenv("MM_FEATUREFLAGS_CUSTOMPROFILEATTRIBUTES", "true")
+	defer os.Unsetenv("MM_FEATUREFLAGS_CUSTOMPROFILEATTRIBUTES")
+	th := Setup(t)
+	defer th.TearDown()
+
+	t.Run("a user without admin permissions should not be able to delete a field", func(t *testing.T) {
+		field := &model.PropertyField{
+			Name: model.NewId(),
+			Type: model.PropertyFieldTypeText,
+		}
+		createdField, _, err := th.SystemAdminClient.CreateCPAField(context.Background(), field)
+		require.NoError(t, err)
+		require.NotNil(t, createdField)
+
+		resp, err := th.Client.DeleteCPAField(context.Background(), createdField.ID)
+		CheckForbiddenStatus(t, resp)
+		require.Error(t, err)
+	})
+
+	th.TestForSystemAdminAndLocal(t, func(t *testing.T, client *model.Client4) {
+		field := &model.PropertyField{
+			Name: model.NewId(),
+			Type: model.PropertyFieldTypeText,
+		}
+		createdField, _, err := th.SystemAdminClient.CreateCPAField(context.Background(), field)
+		require.NoError(t, err)
+		require.NotNil(t, createdField)
+		require.Zero(t, createdField.DeleteAt)
+
+		resp, err := client.DeleteCPAField(context.Background(), createdField.ID)
+		CheckOKStatus(t, resp)
+		require.NoError(t, err)
+
+		deletedField, appErr := th.App.GetCPAField(createdField.ID)
+		require.Nil(t, appErr)
+		require.NotZero(t, deletedField.DeleteAt)
+	}, "a user with admin permissions should be able to delete the field")
+}
+
+func TestListCPAValues(t *testing.T) {
+	os.Setenv("MM_FEATUREFLAGS_CUSTOMPROFILEATTRIBUTES", "true")
+	defer os.Unsetenv("MM_FEATUREFLAGS_CUSTOMPROFILEATTRIBUTES")
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	th.RemovePermissionFromRole(model.PermissionViewMembers.Id, model.SystemUserRoleId)
+	defer func() {
+		th.AddPermissionToRole(model.PermissionViewMembers.Id, model.SystemUserRoleId)
+	}()
+
+	field := &model.PropertyField{
+		Name: model.NewId(),
+		Type: model.PropertyFieldTypeText,
+	}
+	createdField, appErr := th.App.CreateCPAField(field)
+	require.Nil(t, appErr)
+	require.NotNil(t, createdField)
+
+	values := map[string]string{}
+	values[createdField.ID] = "Field Value"
+	_, err := th.Client.PatchCPAValues(context.Background(), values)
+	require.NoError(t, err)
+
+	t.Run("any team member should be able to list values", func(t *testing.T) {
+		th.LoginBasic2()
```

Comment:
This login is needed for both tests here, right? Should we move it outside of the `t.Run`s? Or repeat it in the second one?

Reply:
Good catch! PTAL 294e0eeb90